src/java/org/apache/cassandra/db/guardrails/Threshold.java

```diff
@@ -56,15 +61,15 @@ private String errMsg(boolean isWarning, String what, long value, long threshold
                                              thresholdValue);
     }
 
-    private long abortValue(Config config)
+    private long abortValue(ClientState state)
     {
-        long abortValue = config.getAbortThreshold();
+        long abortValue = abortThreshold.applyAsLong(state);
```

Comment:
Note that in the close future we will have different config objets with different getters. For example, we will have configs for guardrails over the size of a column. The config getters will have different names and signatures, so we don't want to directly call them inside `Threshold`, unless we want to have a different implementation of `Threshold` per type of config.

I thing it's just easier to build `Threshold` with `ToLongFunction`, and make the conversion from the particular config class to the `ToLongFunction` in the `Guardrails` entry point. That way `Threshold` remains simple and decoupled from configuration.

Reply:
Future configs will have different methods with different signatures, for example:
* `int getWarnThreshold()`
* `long getWarnThreshold()`
* `long getWarnThresholdInKB()`
* `DataStorageSpec getWarnThreshold()`
Additionally, some config object might not have one of the thresholds, like the ones for compaction where there isn't an abort threshold. 

How would we know what config getter method we should call here? I think we can't easily have a generic method in `Threshold.Config` because it would mess with reflection, so I guess we should have a separate constructor or factory method for each type of config. I think it's just cleaner to isolate `Threshold` and `Values` from config stuff, and build the generic long providers in `Guardrails`. That way `Threshold` and `Values` would remain unchanged every time we add a new type of config.